docs/operational_guide/bootstrapping.md

```diff
@@ -0,0 +1,112 @@
+# Bootstrapping
+
+## Introduction
+
+We recommend reading the [placement operational guide](placement.md) before reading the rest of this document.
+
+When an M3DB node is turned on (or experiences a placement change) it needs to go through a bootstrapping process to determine the integrity of data that it has, replay writes from the commit log, and/or stream missing data from its peers. In most cases, as long as you're running with the default and recommended bootstrapper configuration of: `filesystem,commitlog,peers,uninitialized_topology` then you should not need to worry about the bootstrapping process at all and M3DB will take care of doing the right thing such that you don't lose data and consistency guarantees are met. Note that the order of the configured bootstrappers *does* matter.
+
+Generally speaking, we recommend that operators do not modify the bootstrappers configuration, but in the rare case that you to, this document is designed to help you understand the implications of doing so.
+
+M3DB currently supports 5 different bootstrappers:
+
+1. `filesystem`
+2. `commitlog`
+3. `peers`
+4. `uninitialized_topology`
+5. `noop_all`
+
+When the bootstrapping process begins, M3DB nodes need to determine two things:
+
+1. What shards they should bootstrap, which can be determined from the cluster placement.
+2. What time-ranges they need to bootstrap those shards for, which can be determined from the namespace retention.
+
+For example, imagine a M3DB node that is responsible for shards 1, 5, 13, and 25 according to the cluster placement. In addition, it has a single namespace called "metrics" with a retention starting 48 hours ago and ending at the current time. When the M3DB node is started, the node will determine that it needs to bootstrap shards 1, 5, 13, and 25 for the time range starting at the current time and ending 48 hours ago. In order to obtain all this data, it will run the configured bootstrappers in the specified order. Every bootstrapper will notify the bootstrapping process of which shard/ranges it was able to bootstrap and the bootstrapping process will continue working its way through the list of bootstrappers until all the shards/ranges required have been marked as fulfilled. Otherwise the M3DB node will fail to start.
```

Comment:
in reference to the comment above, for e.g. this paragraph uses M3DB node to mean M3DB process